lib/array_intersection.rb

```diff
@@ -1,6 +1,24 @@
 # Returns a new array to that contains elements in the intersection of the two input arrays
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) to hash in, O(m) to look up, where n is length of array1 and m is length of array 2
```

Comment:
I would characterize this as O(n + m) where n is the length array1 and m is the length of array2, since you have to loop through each array once.